xds/internal/balancer/edsbalancer/balancergroup_test.go

```diff
@@ -568,3 +582,416 @@ func TestBalancerGroup_start_close_deadlock(t *testing.T) {
 
 	bg.start()
 }
+
+func replaceDefaultSubBalancerCloseTimeout(n time.Duration) func() {
+	old := defaultSubBalancerCloseTimeout
+	defaultSubBalancerCloseTimeout = n
+	return func() { defaultSubBalancerCloseTimeout = old }
+}
+
+// Test that if a sub-balancer is removed, and re-added within close timeout,
+// the subConns won't be re-created.
+func TestBalancerGroup_locality_caching(t *testing.T) {
```

Comment:
This and the two tests are this have the same setup code where they create a `balancerGroup`, add two `subBalancer`s to it, push some `resolvedAddrs`, push some `subConn` state changes, wait for the `picker` updates etc. Could you move all that boiler plate to a helper function which is called from each of these tests. Thanks.

And also, could you please remove the underscores from the test names. And also remove `BalancerGroup` from the test name. So, this test would just become `TestLocalityCaching`.

Reply:
Done